File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/repository/StudentDao.java
package com.mstoyanov.musiclessons.repository;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.mstoyanov.musiclessons.model.Student;

import java.util.List;

@Dao
public interface StudentDao {

    @Query("SELECT * FROM student")
    List<Student> findAll();

    @Insert
    long insert(Student student);

    @Update
    void update(Student student);

    @Delete
    void delete(Student student);
}

File: music-lessons-2-kotlin/app/src/main/java/com/mstoyanov/musiclessons/repository/StudentDao.kt
package com.mstoyanov.musiclessons.repository

import android.arch.persistence.room.*
import com.mstoyanov.musiclessons.model.Student

@Dao
interface StudentDao {

    @Query("SELECT * FROM student")
    fun findAll(): MutableList<Student>

    @Insert
    fun insert(student: Student): Long

    @Update
    fun update(student: Student)

    @Delete
    fun delete(student: Student)
}